src/Sentry.Profiling/SampleProfilerSession.cs

```diff
@@ -1,70 +1,85 @@
+using Microsoft.Diagnostics.Tracing.Etlx;
+using Microsoft.Diagnostics.Tracing.EventPipe;
 using Microsoft.Diagnostics.Tracing.Parsers;
+using Sentry.Extensibility;
+using Sentry.Internal;
 
 namespace Sentry.Profiling;
 
-internal class SampleProfilerSession
+internal class SampleProfilerSession : IDisposable
 {
     private readonly EventPipeSession _session;
-    private readonly MemoryStream _stream;
-    private readonly Task _copyTask;
+    private readonly TraceLogEventSource _eventSource;
+    private readonly SampleProfilerTraceEventParser _sampleEventParser;
+    private readonly IDiagnosticLogger? _logger;
+    private readonly SentryStopwatch _stopwatch;
+    private bool _stopped = false;
 
-    private readonly CancellationTokenRegistration _stopRegistration;
-    private bool _stopped;
-
-    private SampleProfilerSession(EventPipeSession session, MemoryStream stream, Task copyTask, CancellationTokenRegistration stopRegistration)
+    private SampleProfilerSession(SentryStopwatch stopwatch, EventPipeSession session, TraceLogEventSource eventSource, IDiagnosticLogger? logger)
     {
         _session = session;
-        _stream = stream;
-        _copyTask = copyTask;
-        _stopRegistration = stopRegistration;
+        _logger = logger;
+        _eventSource = eventSource;
+        _sampleEventParser = new SampleProfilerTraceEventParser(_eventSource);
+        _stopwatch = stopwatch;
     }
 
     // Exposed only for benchmarks.
     internal static EventPipeProvider[] Providers = new[]
     {
         // Note: all events we need issued by "DotNETRuntime" provider are at "EventLevel.Informational"
         // see https://learn.microsoft.com/en-us/dotnet/fundamentals/diagnostics/runtime-events
-        new EventPipeProvider("Microsoft-Windows-DotNETRuntime", EventLevel.Informational, (long) ClrTraceEventParser.Keywords.Default),
-        new EventPipeProvider("Microsoft-DotNETCore-SampleProfiler", EventLevel.Informational),
-        new EventPipeProvider("System.Threading.Tasks.TplEventSource", EventLevel.Informational, (long) TplEtwProviderTraceEventParser.Keywords.Default)
+        // TODO replace Keywords.Default with a subset. Currently it is:
+        //   Default = GC | Type | GCHeapSurvivalAndMovement | Binder | Loader | Jit | NGen | SupressNGen
+        //                | StopEnumeration | Security | AppDomainResourceManagement | Exception | Threading | Contention | Stack | JittedMethodILToNativeMap
+        //                | ThreadTransfer | GCHeapAndTypeNames | Codesymbols | Compilation,
+        new EventPipeProvider(ClrTraceEventParser.ProviderName, EventLevel.Informational, (long) ClrTraceEventParser.Keywords.Default),
+        new EventPipeProvider(SampleProfilerTraceEventParser.ProviderName, EventLevel.Informational),
+        // new EventPipeProvider(TplEtwProviderTraceEventParser.ProviderName, EventLevel.Informational, (long) TplEtwProviderTraceEventParser.Keywords.Default)
```

Comment:
This isnt' needed now because there's no file involved?
```suggestion
```

Reply:
There were some experiments with this but it didn't give that much value in the past. However, the idea is that it would become necessary for [profiling individual activities](#2316). As is, it cannot be used because the "activity computer" in perfview (that was used to handle these events) isn't yet updated to support streaming event pipe. It would have made the perfview PR too large to hope for a merge so I didn't do it yet. Hoping to get back to it once all this shakes down.